### ISSUE TITLE
Proposal: Adding support for deploying

### ISSUE THREAD

**Issue by jmezach** [jmezach/MSBuild.Sdk.SqlProj#32]
I've been considering adding support for deploying a project that uses this SDK to a SQL Server as part of a `dotnet publish`. I have experimented with this and it is fairly easy to extend the publishing pipeline by adding a `Publish` target to the `Sdk.targets` file and do something from there. But before I go and implement this I was wondering if this would be considered useful for other folks using this package.

I do see a couple of challenges. First of all, we will need to have `sqlpackage.exe` to do the actual publishing. On Windows you get this as part of the SSDT tools, but on Linux and Mac there isn't an easy way to acquire it. You would need to download it and then make it available on the path for this package to use it. Or we could include it in this package, but I'm not sure if we're even allowed to redistribute it. Ideally `sqlpackage.exe` would be available as a .NET Core Global Tool so we could run `dotnet tool install SqlPackage` but that is not how it has been implemented today. Maybe @dzsquared knows if this is changing at some point?

Of course we'd also need to know the server and database name to deploy to. We could probably use localhost and the name of the project as sane defaults, but allow these to be overridden through properties in the project file.

So basically I'm looking for some feedback on this idea. @ErikEJ What do you think?

**Comment by ErikEJ** [comment]
Why do we need sqlpackage(.exe) ? 

https://docs.microsoft.com/en-us/dotnet/api/microsoft.sqlserver.dac.dacservices.deploy?view=sql-dacfx-150 

(Not claiming it will be simple to support all options)

**Comment by jmezach** [comment]
Hmm, yeah, you're right, we don't need to use SqlPackage.exe to deploy it, we could just build a tool ourselves that does it and then run that. Would be quite a bit of work to support all options yes, but maybe we don't need to support them all since it will be mostly used for local development I think.

Maybe we need to rename the current BuildDacpac tool and add the support there, or have a separate tool. They have a different purpose of course, but maybe including yet another tool in the same package is a bit too much overhead. We would be duplicating dependencies in that case.

**Comment by jmezach** [comment]
This has been implemented in #33 and should be released soon as 1.2.0. I'll leave this open until that version is out.

**Comment by jmezach** [comment]
[1.2.0](https://www.nuget.org/packages/MSBuild.Sdk.SqlProj/1.2.0) is out on NuGet.org which has support for this. Thanks @ErikEJ for reviewing the change.
